cmake/install/InstallTargetDependencies.cmake

```diff
@@ -0,0 +1,74 @@
+# Like INSTALL_DEPENDENCIES but can be called from regular cmake code
+# (instead of install(CODE)), takes targets instead of files,
+# takes care of configuring search paths, and other platform-specific tweaks.
+FUNCTION(INSTALL_TARGET_DEPENENCIES)
+    set(options)
+    set(oneValueArgs NAME)
+    set(multiValueArgs TARGETS DESTINATION LIB_DIRS_SUFFIXES LIB_DIRS)
+    cmake_parse_arguments(DEPS "${options}" "${oneValueArgs}"
+            "${multiValueArgs}" ${ARGN} )
+
+    if(NOT DEPS_LIB_DIRS)
+        set(DEPS_LIB_DIRS "")
+    endif()
+
+    FOREACH(TARGET ${DEPS_TARGETS})
+        IF(NOT TARGET ${TARGET})
+            message(FATAL_ERROR "Not a target: ${TARGET}")
+        ENDIF()
+
+        # Collect target output files.
+        LIST(APPEND DEPLOY_TARGETS "$<TARGET_FILE:${TARGET}>")
+
+        # Collect target link directories
+        get_target_property(target_libs ${TARGET} LINK_LIBRARIES)
+
+        foreach(lib ${target_libs})
+            if(TARGET ${lib} OR NOT IS_ABSOLUTE ${lib})
+                continue()
+            endif()
+
+            get_filename_component(lib_dir ${lib} PATH)
+            list(APPEND DEPS_LIB_DIRS ${lib_dir})
+        endforeach()
+    ENDFOREACH()
+
+    FOREACH(LIB_PATH ${DEPS_LIB_DIRS})
+        FOREACH(suffix ${DEPS_LIB_DIRS_SUFFIXES})
+            list(APPEND DEPS_LIB_DIRS "${LIB_PATH}/${suffix}")
+        ENDFOREACH()
+    ENDFOREACH()
+
+    # Create the list of files using file(GENERATE)
+    # generator expressions on install(CODE) are not supported on old
+    # cmake versions.
+    SET(DEPLOY_LIST_FILE "${CMAKE_CURRENT_BINARY_DIR}/${DEPS_NAME}_filelist_${CMAKE_BUILD_TYPE}.txt")
```

Comment:
> it probably has something to do with those generator expressions.
```suggestion
    SET(DEPLOY_LIST_FILE "${CMAKE_CURRENT_BINARY_DIR}/${DEPS_NAME}_filelist_$<CONFIG>.txt")
```

Reply:
> I don't get it; for some strange reason cmake tries to write each `filelists.txt` 9 times with appveyor.

Overwriting the file isn't a problem. The real problem is the content differs in Release and Debug configuration.
https://github.com/LMMS/lmms/blob/ce5c74432e83f0eaa617cc26332f4209bc4efc7f/cmake/modules/BuildPlugin.cmake#L75-L78
The `SET_TARGET_PROPERTIES` shouldn't check for `STRIP`.